lib/srv/discovery/discovery.go

```diff
@@ -1656,7 +1656,7 @@ func (s *Server) upsertDynamicMatchers(ctx context.Context, dc *discoveryconfig.
 	s.dynamicServerGCPFetchers[dc.GetName()] = gcpServerFetchers
 	s.muDynamicServerGCPFetchers.Unlock()
 
-	databaseFetchers, err := s.databaseFetchersFromMatchers(matchers)
+	databaseFetchers, err := s.databaseFetchersFromMatchers(matchers, dc.GetName())
```

Comment:
just curious what happens when multiple discovery configs match the same database?

Reply:
If running on the same DiscoveryService (or same `discovery_group`), it will double the API calls, but we should get consistent states (no duplicated Databases or missing ones).